client/api/Config.cs

```diff
@@ -1,85 +1,78 @@
-﻿using System;
+using System;
 using io.harness.cfsdk.client.cache;
+using Microsoft.Extensions.Logging;
 
 namespace io.harness.cfsdk.client.api
 {
     public class Config
     {
-        public static int MIN_FREQUENCY = 60;
+        public static readonly int MIN_FREQUENCY = 60;
 
-        public string ConfigUrl { get => configUrl; }
-        internal string configUrl = "https://config.ff.harness.io/api/1.0";
-        public string EventUrl { get => eventUrl; }
-        internal string eventUrl = "https://events.ff.harness.io/api/1.0";
-        public bool StreamEnabled { get => streamEnabled; }
-        internal bool streamEnabled = true;
+        public string ConfigUrl { get; internal set; } = "https://config.ff.harness.io/api/1.0";
+        public string EventUrl { get; internal set; } = "https://events.ff.harness.io/api/1.0";
+        public bool StreamEnabled { get; internal set; } = true;
 
-        public int PollIntervalInMiliSeconds { get => pollIntervalInSeconds * 1000; }
-        internal int pollIntervalInSeconds = 60;
+        public int PollIntervalInSeconds { get; internal set; } = 60;
+        public int PollIntervalInMiliSeconds => PollIntervalInSeconds * 1000;
 
         // configurations for Analytics
-        public bool AnalyticsEnabled { get => analyticsEnabled; }
-        internal bool analyticsEnabled = true;
+        public bool AnalyticsEnabled { get; internal set; } = true;
 
         public int Frequency { get => Math.Max(frequency, Config.MIN_FREQUENCY); }
         private int frequency = 60;
 
-        public ICache Cache { get => cache; }
-        internal ICache cache = new FeatureSegmentCache();
+        public ICache Cache { get; internal set; } = new FeatureSegmentCache();
 
-        public IStore Store { get => store;  }
-        internal IStore store = null;
+        public IStore Store { get; internal set; } = null;
 
-        //BufferSize must be a power of 2 for LMAX to work. This function vaidates
-        //that. Source: https://stackoverflow.com/a/600306/1493480
         public int BufferSize
         {
-            get
+            get { return bufferSize; }
+            internal set
             {
-                if (!(bufferSize != 0 && ((bufferSize & (bufferSize - 1)) == 0)))
-                {
+                // BufferSize must be a positive power of 2 for LMAX to work. 
+                // Source: https://stackoverflow.com/a/600306/1493480
+                if (value <= 0 || (value & (value - 1)) != 0)
```

Comment:
Moved this check to the setter instead of the getter. Also check for negative values.